### PR TITLE
Automatically apply database migrations via CI

### LINKED ISSUE #193: Handle automatic migrations in production

### PR DESCRIPTION
**Pull request by kevincianfarini**
Closes #193

### CODE REVIEW

File: backend/src/main/sqldelight/migrations/V1__drop_phone_number_called.sqm [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE action_call_legislator
```
Comment (kevincianfarini) [inline backend/src/main/sqldelight/migrations/V1__drop_phone_number_called.sqm:1]: These migrations were altered to make them repeatable. Applying them to a database which have already executed these migrations will do nothing.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -49,3 +50,20 @@ publish-production-container:
...
+  image: 
+    name: flyway/flyway:8.2.2-alpine
+    entrypoint: [""]
+  script:
+    # below is a hack to work around https://github.com/cashapp/sqldelight/issues/2744. Can remove manual file renaming when it's resolved. 
+    - mkdir -p backend/build/resources/main/migrations/
+    - cp backend/src/main/sqldelight/migrations/*.sqm backend/build/resources/main/migrations/
+    - find backend/build/resources/main/migrations/ -name "*.sqm" -exec sh -c 'mv "$1" "${1%.sqm}.sql"' _ {} \;
+    - flyway -user=${POSTGRES_USER} -password=${POSTGRES_PASSWORD} -table=schema_history -locations=./backend/build/resources/main/migrations/ -url=${POSTGRES_URL} migrate
+  only:
```
Comment (kevincianfarini) [inline .gitlab-ci.yml:65]: This CI job will only run when the target is a commit on `trunk` _and_ when there's changes to any of the migrations.

**Comment by kevincianfarini** [comment]
When this PR gets merged into trunk, the migration job should run, _however_ it will be a no-op. I've baselined the production database at version 2, which means there's no actual migrations to apply. 

```
ccm_readonly@ccm-db:postgres> select * from schema_history;
+------------------+-----------+-----------------------+----------+-----------------------+------------+----------------+----------------------------+------------------+-----------+
| installed_rank   | version   | description           | type     | script                | checksum   | installed_by   | installed_on               | execution_time   | success   |
|------------------+-----------+-----------------------+----------+-----------------------+------------+----------------+----------------------------+------------------+-----------|
| 1                | 2         | << Flyway Baseline >> | BASELINE | << Flyway Baseline >> | <null>     | null           | 2021-12-22 17:23:28.340378 | 0                | True      |
+------------------+-----------+-----------------------+----------+-----------------------+------------+----------------+----------------------------+------------------+-----------+
SELECT 1
Time: 0.101s
ccm_readonly@ccm-db:postgres>
```

File: backend/src/main/sqldelight/migrations/V1__drop_phone_number_called.sqm (hunk 1 above)
Comment (EvanHahn) [reply backend/src/main/sqldelight/migrations/V1__drop_phone_number_called.sqm:1]: Could be worth adding this GitHub comment as a "real" in-code comment, as it seems useful for the future.

**Review by EvanHahn — APPROVED** [review]

**Pull request merged**